source/s3_auto_ranged_put.c

```diff
@@ -1220,6 +1010,27 @@ static void s_s3_prepare_upload_part_on_read_done(void *user_data) {
         aws_s3_meta_request_unlock_synced_data(meta_request);
     }
     /* END CRITICAL SECTION */
+    if (uploaded_part_info) {
+        /* Check for the durability */
```

Comment:
Adding explainer about why it's safe to touch this after dropping the lock (copy pasted from when we did a similar thing in the skipping logic)
```suggestion
    /* END CRITICAL SECTION */
    
    if (previously_uploaded_part) {
        /* Part was previously uploaded, check that it matches what we just read.
         * (Yes it's weird that we keep a pointer to the part_info even after
         * releasing the lock that protects part_list. But it's the resizable
         * part_list that needs lock protection. A previously uploaded part_info is const,
         * and it's on the heap, so it's safe to keep the pointer around)
```